### PR TITLE
Use single-task queues in ReservedClusterStateService

### PR DESCRIPTION
**Pull request by prdoyle**
These tasks don't support batching, so submit each in its own queue.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java [hunk 1]
```diff
@@ -234,15 +226,15 @@ public void process(
...
+        submitUpdateTask(
             "reserved cluster state [" + namespace + "]",
             new ReservedStateUpdateTask(
                 namespace,
                 reservedStateChunk,
                 versionCheck,
                 handlers,
                 orderedHandlers,
-                ReservedClusterStateService.this::updateErrorState,
+                this::updateErrorState,
```
Comment (prdoyle) [inline server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java:237]: Huh, IntelliJ did this automatically.

File: server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java [hunk 2]
```diff
@@ -261,9 +253,17 @@ public void onFailure(Exception e) {
...
-            null
+            )
+        );
+    }
+
+    private void submitUpdateTask(String source, ReservedStateUpdateTask task) {
+        var updateTaskQueue = clusterService.createTaskQueue(
+            "reserved state update",
+            Priority.URGENT,
+            new ReservedStateUpdateTaskExecutor(rerouteService)
```
Comment (prdoyle) [inline server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java:264]: Wait this doesn't look right. We just want one executor I think.

**Review by DaveCTurner — COMMENTED** [review]
Yep that should work. Ideally we'd have some test to show that no batching happens tho.

**Comment by prdoyle** [comment]
Sorry for the force-push. I had a lot of messing around and wanted to leave the result in a reviewable state.

This branch now has three commits:
1. A refactoring to expose a method I want to call from the tests
2. The tests. They fail without the fix.
3. The fix. These make the tests pass.

You might want to review this PR commit-by-commit.

**Review by DaveCTurner — APPROVED** [review]
LGTM (although bleurgh more mocks)

**Pull request merged**